### PR TITLE
validation: add CEL validation to workload entry

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: networking/v1alpha3/service_entry.proto [hunk 1]
```diff
@@ -554,6 +554,7 @@ message ServiceEntry {
 
   // One or more endpoints associated with the service. Only one of
   // `endpoints` or `workloadSelector` can be specified.
+  // +kubebuilder:validation:MaxItems=4096
```
Comment (hzxuzhonghu) [inline networking/v1alpha3/service_entry.proto:557]: If we allow 4096 endpoints, then the workloadEntry must exceed the max size limit of a object from etcdm, which is likely 1.5MB

Comment (howardjohn) [reply networking/v1alpha3/service_entry.proto:557]: right now we allow infinity. So this is only making it more restrictive, not less

Comment (hzxuzhonghu) [reply networking/v1alpha3/service_entry.proto:557]: I know, could we strict limit it.  In real scenario, can limit to 128. If more endpoints associated, we should encourage to use workloadentry separately

Comment (howardjohn) [reply networking/v1alpha3/service_entry.proto:557]: If I was building today I would just do not want to break existing users if they have a lot?

**Review by hzxuzhonghu — APPROVED** [review]

File: networking/v1alpha3/service_entry.proto (hunk 1 above)
Comment (nrjpoddar) [inline networking/v1alpha3/service_entry.proto:557]: We should add documentation around it too. I don't expect folks to have so many entries but we should document our limits clearly.

**Review by nrjpoddar — COMMENTED** [review]
LGTM

**Review by linsun — APPROVED** [review]
LGTM

**Pull request merged**